enterprise_access/apps/content_assignments/content_metadata_api.py

```diff
@@ -48,18 +31,9 @@ def get_content_metadata_for_assignments(enterprise_catalog_uuid, assignments):
         to a content metadata dictionary, or null if no such dictionary
         could be found for a given key.
     """
-    is_run_based_by_content_key = {
-        assignment.content_key: assignment.is_assigned_course_run
-        for assignment in assignments
-    }
-    content_metadata_list = get_and_cache_catalog_content_metadata(
-        enterprise_catalog_uuid,
-        is_run_based_by_content_key.keys()
-    )
-    metadata_by_key = {}
-    for course_metadata in content_metadata_list:
-        _process_course_metadata(course_metadata, is_run_based_by_content_key, metadata_by_key)
-
+    content_keys = {assignment.content_key for assignment in assignments}
+    content_metadata_list = get_and_cache_catalog_content_metadata(enterprise_catalog_uuid, content_keys)
+    metadata_by_key = dict(zip(content_keys, content_metadata_list))
```

Comment:
[inform] Hm, actually... still noodling as this approach with `zip` assumes the same ordering between `content_keys` and `content_metadata_list`. I was trying to avoid what would have otherwise been nested `for` loops.